File: app/src/main/java/com/tutv/android/di/Container.java
package com.tutv.android.di;

import android.content.SharedPreferences;

import com.tutv.android.repository.SeriesRepository;
import com.tutv.android.repository.UserRepository;
import com.tutv.android.utils.schedulers.BaseSchedulerProvider;

public interface Container {
    UserRepository getUserRepository();
    SeriesRepository getSeriesRepository();
    BaseSchedulerProvider getSchedulerProvider();
}

File: app/src/main/java/com/tutv/android/di/Container.kt
package com.tutv.android.di

import com.tutv.android.repository.SeriesRepository
import com.tutv.android.repository.UserRepository
import com.tutv.android.utils.schedulers.BaseSchedulerProvider

interface Container {
    val userRepository: UserRepository
    val seriesRepository: SeriesRepository
    val schedulerProvider: BaseSchedulerProvider
}